src/js/tech/html5.js

```diff
@@ -1319,7 +1319,7 @@ const mp4RE = /^video\/mp4/i;
 
 Html5.patchCanPlayType = function() {
   // Android 4.0 and above can play HLS to some extent but it reports being unable to do so
-  if (browser.ANDROID_VERSION >= 4.0) {
+  if (browser.ANDROID_VERSION >= 4.0 && !browser.IS_FIREFOX) {
```

Comment:
The support in Firefox 50 is not polished, there are a lot of bugs (e.g. https://bugzilla.mozilla.org/show_bug.cgi?id=1301050). It will probably be disabled in 50.
We're using a hacky solution, so it will never be like a properly supported codec.
I would leave it as is and expect Firefox not to return wrong values in `canPlayType`, if it will ever support HLS natively, it will be returned in `canPlayType`.